lib/hanami/router.rb

```diff
@@ -60,13 +61,14 @@ def self.define(&blk)
     #   Hanami::Router.new do
     #     get "/", to: ->(*) { [200, {}, ["OK"]] }
     #   end
-    def initialize(base_url: DEFAULT_BASE_URL, prefix: DEFAULT_PREFIX, resolver: DEFAULT_RESOLVER, block_context: nil, &blk)
+    def initialize(base_url: DEFAULT_BASE_URL, prefix: DEFAULT_PREFIX, resolver: DEFAULT_RESOLVER, block_context: nil, default_app: not_found, &blk)
```

Comment:
@katafrakt I'd rename the param to `not_found:`.

Reply:
@katafrakt Please pass the default as `NOT_FOUND` (a lambda that returns the default 404 response), and let the `not_found` method to simply invoke `@not_found`.